aws_lambda_builders/workflows/go_dep/workflow.py

```diff
@@ -0,0 +1,63 @@
+"""
+Go Dep Workflow
+"""
+
+import logging
+import os
+
+from aws_lambda_builders.actions import CopySourceAction
+from aws_lambda_builders.workflow import BaseWorkflow, Capability
+
+from .actions import DepEnsureAction, GoBuildAction
+from .utils import OSUtils
+from .subproc_exec import SubprocessExec
+
+LOG = logging.getLogger(__name__)
+
+class GoDepWorkflow(BaseWorkflow):
+    """
+    A Lambda builder workflow that knows how to build
+    Go projects using `dep`
+    """
+
+    NAME = "GoDepBuilder"
+
+    CAPABILITY = Capability(language="go",
+                            dependency_manager="dep",
+                            application_framework=None)
+
+    EXCLUDED_FILES = (".aws-sam")
+
+    def __init__(self,
+                source_dir,
+                artifacts_dir,
+                scratch_dir,
+                manifest_path,
+                runtime=None,
+                osutils=None,
+                **kwargs):
+
+        super(GoDepWorkflow, self).__init__(source_dir,
+                                            artifacts_dir,
+                                            scratch_dir,
+                                            manifest_path,
+                                            runtime=runtime,
+                                            **kwargs)
+
+        options = kwargs["options"] if "options" in kwargs else {}
+        handler = options.get("handler", None)
+
+        if osutils is None:
+            osutils = OSUtils()
+
+        # project base name, where the Gopkg.toml and vendor dir are.
+        base_dir = osutils.abspath(osutils.dirname(manifest_path))
+        output_path = osutils.joinpath(osutils.abspath(artifacts_dir), handler)
+
+        subprocess_dep = SubprocessExec(osutils, "dep")
+        subprocess_go = SubprocessExec(osutils, "go")
+
+        self.actions = [
```

Comment:
We don't need to copy the source anywhere for this to work?

Reply:
Amazingly, no. we don't change anything for the source, other than ensuring we're have the deps, which the user will probably want anyway.

The version where we copy source was rather complex for the sake of it.